rts/Rml/Backends/RmlUi_Backend.cpp

```diff
@@ -364,73 +354,100 @@ bool RmlGui::ProcessMouseEvent(const SDL_Event &event)
 		return false;
 	}
 	}
+	return false;
 }
 
 /*
 	Return true if the event was handled by rmlui
 */
-bool RmlGui::ProcessKeyPressed(int keyCode, int scanCode, bool isRepeat)
+bool RmlGui::ProcessMouseEvent(const SDL_Event &event)
 {
-	if (!data->initialized || data->contexts.size() < 1)
+	if (data == NULL || data->contexts.size() < 1 || !data->initialized)
 	{
 		return false;
 	}
-	auto kc = RmlSDL::ConvertKey(keyCode);
-	// if (kc == Rml::Input::KI_ESCAPE)
-	// {
-	// 	Rml::Element *el = data->contexts[0]->GetFocusElement();
-	// 	if (el)
-	// 	{
-	// 		el->Blur();
-	// 	}
-	// 	return false;
-	// }
-
-	return !RmlSDL::EventKeyDown(data->contexts[0], kc);
+	bool result = false;
+	for (auto &context : data->contexts)
+	{
+		result |= processContextMouseEvent(context, event);
```

Comment:
should all contexts get the mouse event? shouldn't it stop iterating if one consumes it?

Reply:
What if it is a click, then the other contexts need to know to unfocus any text input boxes, to unhover elements, etc.

Reply:
What happens if two windows with a button overlap such that both buttons are under the mouse? Like this
![image](https://github.com/beyond-all-reason/spring/assets/2573076/dba6244c-d069-47cd-84f9-f16422fc5d15)

Unhover/unfocus is a good point. The way the Chili framework solves it is that it keeps a reference to the currently hovered/focused element (should be a single one in each case), have the first element that accepts the event consume it (stopping further iteration), and if this context is a different one than the one that previously had focus then send a "got unfocused" event to the old one directly.

Reply:
I would just recommend not using 2 contexts that occupy the same area. Contexts are instead used for multiple monitors, or perhaps a separate context for LuaMenu vs LuaUi, where one is hidden when the other is activated. Multiple draggable windows would all be in the same rmlui context

Reply:
If screens are separate contexts, does that mean I can't drag a draggable panel/window across screens? If I press Esc to close the most recently opened panel, will it also close some panel on the other screen since it independently receives that event? And if I press Tab to switch textbox focus, will each screen cycle them independently so I get two focuses, then get to type in both because they both receive keyboard events?

Reply:
You are right, multiple monitors would also be a single context. Here is what the documentation says multiple contexts could be useful for: https://mikke89.github.io/RmlUiDoc/pages/cpp_manual/contexts.html#uses-of-multiple-contexts

Reply:
Sounds reasonable.